run.rb

```diff
@@ -1,26 +1,32 @@
-require 'rubygems'
-require 'bundler/setup'
-require 'dropbox_sdk'
+require 'dropbox'
 require 'dotenv'
 Dotenv.load
 
-dropbox = DropboxClient.new(ENV['API_KEY'])
-uploads = dropbox.metadata(ENV['UPLOADS'], 25_000, true)['contents']
+dropbox = Dropbox::Client.new(ENV['API_KEY'])
+uploads = dropbox.list_folder(ENV['UPLOADS']).entries
 
 uploads.each do |upload|
-  unless upload['is_dir'] 
-    folder = upload['path'].match(/[0-9]{4}\-[0-9]{2}/).to_s
-    new_folder = "/#{ENV['UPLOADS']}/#{folder}"
+  unless upload.is_a?(Dropbox::FolderMetadata)
+    puts "Considering #{upload.path_display}"
+
+    folder = upload.path_display.match(/[0-9]{4}\-[0-9]{2}/).to_s
```

Comment:
Did you have some bad examples for this? Or does the upload object return various other objects that don't look like a folder? Is there a boolean on the upload object that indicates it's a folder perhaps?

Reply:
I had some photos that didn't follow a standard naming convention. I introduced this conditional check to skip over any files that didn't match the regex.